toolchain/docs/check/pattern_matching.md

```diff
@@ -0,0 +1,285 @@
+# Pattern matching
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Glossary](#glossary)
+-   [Overview](#overview)
+-   [Pattern insts](#pattern-insts)
+-   [Inst ordering](#inst-ordering)
+-   [Parser-driven pattern block pushing](#parser-driven-pattern-block-pushing)
+-   [Function parameters](#function-parameters)
+    -   [`Call` parameters and arguments](#call-parameters-and-arguments)
+    -   [Caller and callee matching](#caller-and-callee-matching)
+    -   [The return slot](#the-return-slot)
+
+<!-- tocstop -->
+
+## Glossary
+
+A _full-pattern_ is a pattern that is not a subpattern of an enclosing pattern.
```

Comment:
This seems different from the [definition in variadics](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/design/variadics.md):

> A _full pattern_ consists of an optional deduced parameter list, a pattern, and
an optional return type expression.

Could that same definition be used here, or is this using the same term for subtly different meanings?

As an aside, noting the main [pattern matching design](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/design/pattern_matching.md#overview) defines scrutinee, perhaps it should define for "full pattern" too?

Reply:
Good point! I've moved this definition and the one in `variadics.md` to `design/pattern_matching.md`.